src/apps/external_executor/external_executor.cpp

```diff
@@ -45,6 +52,35 @@ namespace externalexecutor
         ccf::no_auth_required)
         .set_auto_schema<void, GetExecutorCode::Out>()
         .install();
+
+      // create an endpoint to register the executor
+      auto register_executor = [this](auto& ctx, ccf::NewExecutor&& payload)
+        -> ccf::grpc::GrpcAdapterResponse<ccf::RegistrationResult> {
+        // generate and store executor node id locally
+        crypto::Pem executor_public_key(payload.cert());
+        auto pubk_der = crypto::cert_pem_to_der(executor_public_key);
+        ExecutorNodeId executor_node_id =
+          ccf::compute_node_id_from_pubk_der(pubk_der);
+
+        ExecutorNodeInfo executor_info = {
```

Comment:
We should check that the hash of the public key is recorded as report data in the attestation.

Reply:
How difficult is it to create an attestation with specific report data? As in, is this more difficult/more awkward to do in arbitrary languages than getting a 'dumb' attestation?

(I agree we need to do this regardless, just thinking about the implication - if it's tricky it's more likely we build a sidecar/binary/call-this solution)

Reply:
For SEV-SNP, we need to make an `ioctl` call so probably OK for most low-level languages but awkward from something like JS? For SGX, we need to call into Open Enclave so even more awkward. It seems to me that this needs to be a separate binary, either running in the same container as the BL or a side-car as you said.